pyogrio/raw.py

```diff
@@ -472,29 +474,27 @@ def _parse_options_names(xml):
 
 
 def _validate_metadata(dataset_metadata, layer_metadata, metadata):
-    """ """
+    """Validate the metadata."""
     if metadata is not None:
         if layer_metadata is not None:
             raise ValueError("Cannot pass both metadata and layer_metadata")
         layer_metadata = metadata
 
     # validate metadata types
-    for metadata in [dataset_metadata, layer_metadata]:
-        if metadata is not None:
-            for k, v in metadata.items():
+    for meta in [dataset_metadata, layer_metadata]:
```

Comment:
Any particular reason for the change of `metadata` to `meta`?  Either is fine, but `metadata` was slightly more clear.

Reply:
There is already an input parameter called `metadata` in the function definition. The local variable of the for loop redefines this variable, which is something ruff doesn't like because it can be confusing...